src/server/scripts/Zandalar/AtalDazar/boss_priestess_alun_za.cpp

```diff
@@ -0,0 +1,684 @@
+                        });
+
+                        if (closestAt == atList.end())
+                            return;
+
+                        me->SetReactState(REACT_PASSIVE);
+                        me->GetMotionMaster()->MovePoint(POINT_POOL, (*closestAt)->GetPosition());
+                        _events.CancelEvent(EVENT_CHECK_TAINTED_BLOOD);
+                    }
+                    break;
+                }
+                default:
+                    break;
+            }
+
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
+        }
+    }
+
+private:
+    EventMap _events;
+};
+
+// 258388 - Ritual
+class spell_priestess_alun_za_ritual : public SpellScript
+{
+    static void SetDest(SpellDestination& dest)
+    {
+        dest.Relocate(Trinity::Containers::SelectRandomContainerElement(RitualPosition));
+    }
+
+    void Register() override
+    {
+        OnDestinationTargetSelect += SpellDestinationTargetSelectFn(spell_priestess_alun_za_ritual::SetDest, EFFECT_0, TARGET_DEST_DEST);
+    }
+};
+
+// 258681 - Energy Regen
+class spell_priestess_alun_za_energy_regen : public AuraScript
+{
+    void OnPeriodic(AuraEffect const* aurEff) const
+    {
+        Unit* target = GetTarget();
+        if (target)
+            target->ModifyPower(POWER_ENERGY, aurEff->GetAmount() / 10);
+    }
+
+    void Register() override
+    {
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_priestess_alun_za_energy_regen::OnPeriodic, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
+    }
+};
+
+// 255615 - Agitate
+class spell_priestess_alun_za_agitate : public SpellScript
+{
+    void HandleScript(SpellEffIndex /*effIndex*/) const
+    {
+        GetCaster()->CastSpell(GetHitUnit(), GetEffectValue(), CastSpellExtraArgsInit{
+            .TriggerFlags = TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR,
+            .TriggeringSpell = GetSpell(),
+            .OriginalCastId = GetSpell()->m_castId
+        });
+    }
+
+    static void SelectTarget(std::list<WorldObject*>& targets)
+    {
+        targets.resize(1);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_priestess_alun_za_agitate::HandleScript, EFFECT_0, SPELL_EFFECT_DUMMY);
+        OnObjectAreaTargetSelect += SpellObjectAreaTargetSelectFn(spell_priestess_alun_za_agitate::SelectTarget, EFFECT_0, TARGET_UNIT_SRC_AREA_ENTRY);
+    }
+};
+
+// 255591 - Molten Gold
+class spell_priestess_alun_za_molten_gold : public AuraScript
+{
+    void OnRemove(AuraEffect const* aurEff, AuraEffectHandleModes /*mode*/) const
+    {
+        if (GetTargetApplication()->GetRemoveMode() != AURA_REMOVE_BY_EXPIRE)
+            return;
+
+        Unit* target = GetTarget();
+        if (!target)
+            return;
+
+        target->CastSpell(target, GetEffectInfo(EFFECT_0).CalcValue(), CastSpellExtraArgsInit{
```

Comment:
CalcValue would only matter here if there was an aura which modifies the BasePoints of this spell's EFFECT_0

Reply:
bad suggestion, always calcvalue, never read basepoints directly

Reply:
oh I thought we were not supposed to use CalcValue when not required, noted